crypt.go

```diff
@@ -304,7 +305,7 @@ func activateWithRecoveryKey(volumeName, sourceDevicePath string, keyReader io.R
 			continue
 		}
 
-		if _, err := unix.AddKey("user", fmt.Sprintf("%s:%s:reason=%d", filepath.Base(os.Args[0]), volumeName, reason), key[:], userKeyring); err != nil {
+		if _, err := unix.AddKey("user", fmt.Sprintf("secboot:%s?type=recovery&reason=%d", sourceDevicePath, reason), key[:], userKeyring); err != nil {
```

Comment:
this looks much better, but the caller still needs the options of:

* choosing the prefix part of this (secboot vs something more caller specific)
* replace this entirely with something else